int32 nMarkers
marker[] markers